### PR TITLE
Output original Dockerfile when build fails

### PR DESCRIPTION
**Pull request by tkdchen**
* CLOUDBLD-10133

Signed-off-by: Chenxiong Qi <cqi@redhat.com>

# Maintainers will complete the following section

- [ ] Commit messages are descriptive enough
- [ ] Code coverage from testing does not decrease and new code is covered
- [ ] Python type annotations added to new code
- [ ] JSON/YAML configuration changes are updated in the relevant schema
- [ ] Changes to metadata also update the documentation for the metadata
- [ ] Pull request has a link to an osbs-docs PR for user documentation updates
- [ ] New feature can be disabled from a configuration file

### CODE REVIEW

**Comment by chmeliik** [comment]
> Unfortunately, the output of exit tasks will not be in the logs disappointed (at least not until CLOUDBLD-9857 is done, if it is even possible)

Ignore me, it won't be in the Brew **build** logs, but it will be in the Brew **task** logs (those are uploaded by koji-containerbuild) and that's good enough (failed tasks don't even produce builds)

File: tekton/pipelines/binary-container.yaml [hunk 1]
```diff
@@ -287,3 +287,20 @@ spec:
           value: "$(context.pipelineRun.name)"
         - name: user-params
           value: '$(params.user-params)'
+    - name: binary-container-output-dockerfile
+      when:
+        - input: "$(tasks.status)"
+          operator: in
+          values:
+            - Failed
```
Comment (chmeliik) [inline tekton/pipelines/binary-container.yaml:295]: I'd say this doesn't need to be conditional, the original Dockerfile might be useful for successful builds too

(and if it's not conditional, I wonder if it still makes sense to have it as a separate task)

Comment (rcerven) [reply tekton/pipelines/binary-container.yaml:295]: yeah it should be consistent, for both successful & failed builds

and I'd just add it to some exit task, this doesn't even deserve it's own exit plugin (maybe store_metadata)

but also having it as separate task doesn't make sense

Comment (tkdchen) [reply tekton/pipelines/binary-container.yaml:295]: Have opened another PR #1942. This one will be closed.

**Pull request closed without merging**